aws/logs_monitoring/lambda_function.py

```diff
@@ -762,13 +791,22 @@ def normalize_events(events, metadata):
     for event in events:
         events_counter += 1
         if isinstance(event, dict):
-            normalized.append(merge_dicts(event, metadata))
+            normalized_event = merge_dicts(event, metadata)
         elif isinstance(event, str):
-            normalized.append(merge_dicts({"message": event}, metadata))
+            normalized_event = merge_dicts({"message": event}, metadata)
         else:
             # drop this log
             continue
 
+        # in case it's a cloudtrail event replace the hostname with the
+        # hostname in the ARN if available.
+        if normalized_event.get(DD_SOURCE) == "cloudtrail":
+            extracted_arn_hostname = extract_arn_hostname(normalized_event["message"])
```

Comment:
```suggestion
            extracted_arn_hostname = extract_arn_hostname(normalized_event)
```

There is no `message` key. Passing in the normalized_event dict will work just fine.